bundler/lib/bundler/compact_index_client.rb

```diff
@@ -82,10 +82,9 @@ def dependencies(names)
 
     def update_and_parse_checksums!
       Bundler::CompactIndexClient.debug { "update_and_parse_checksums!" }
-      return @info_checksums_by_name if @parsed_checksums
+      return if @parsed_checksums
       update("versions", @cache.versions_path, @cache.versions_etag_path)
-      @info_checksums_by_name = @cache.checksums
-      @parsed_checksums = true
+      @parsed_checksums = !@cache.info_checksum_for("bundler").nil?
```

Comment:
This is the stand in for "is the file parseable?".

Reply:
i dont think this works, there can be other gem sources that don't include bundler

Reply:
Good point.

Reply:
My new approach is to parse the last line, expecting it to end with a newline and have at least one space on the line. If the versions file is malformed after all the work we put in to make sure it's not, then I think it's just going to have to crash and fallback.